openrtb-core/src/main/java/com/iabtechlab/openrtb/v2/json/OpenRtbJsonWriter.java

```diff
@@ -953,6 +959,63 @@ protected void writeGeoFields(Geo geo, JsonGenerator gen) throws IOException {
     }
   }
 
+  public final void writeSua(UserAgent sua, JsonGenerator gen) throws IOException {
```

Comment:
I think it's better call it `UserAgent userAgent`. `sua` is the name of the field, and you specified it above